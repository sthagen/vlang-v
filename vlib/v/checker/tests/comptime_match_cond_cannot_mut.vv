fn main() {
	mut x := 1
	$match x {
		1 {
			println('1')
		}
		2 {
			println('2')
		}
	}

	y := 100
	$match mut y {
		100 {
			println('100')
		}
	}
}
