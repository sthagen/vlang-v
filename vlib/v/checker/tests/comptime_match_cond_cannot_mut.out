vlib/v/checker/tests/comptime_match_cond_cannot_mut.vv:3:9: error: `x` is mut and may have changed since its definition
    1 | fn main() {
    2 |     mut x := 1
    3 |     $match x {
      |            ^
    4 |         1 {
    5 |             println('1')
vlib/v/checker/tests/comptime_match_cond_cannot_mut.vv:13:13: error: `$match` condition `y` can not be mutable
   11 | 
   12 |     y := 100
   13 |     $match mut y {
      |                ^
   14 |         100 {
   15 |             println('100')
